lib/jekyll/commands/publish.rb

```diff
@@ -59,6 +59,10 @@ def to
         date_stamp = params.date.strftime Jekyll::Compose::DEFAULT_DATESTAMP_FORMAT
         "_posts/#{date_stamp}-#{params.name}"
       end
+
+      def front_matter(data)
+        data.merge("date" => params.date.strftime("%Y-%m-%d %H:%M %z"))
```

Comment:
IMO, it would be better if we assign `date` to a data when it is otherwise not available. (*Open for debate*):
```suggestion
        data["date"] ||= params.date.strftime("%Y-%m-%d %H:%M %z"))
        data
```

Reply:
I don't know Ruby well enough to debate, unfortunately. I will go with your suggestion.